### PR TITLE
fix(plan-mode): prevent infinite /plan review loop

### PR DESCRIPTION
**Pull request by FradSer**
## What
Fixes an issue in `@fradser/pi-plan-mode` where completing a plan dispatched `/plan review` as a conversation turn back to the LLM instead of opening the review overlay directly, causing an infinite loop. Also ensures `newSession` command context availability and leading-tilde path expansion.

## Why
When `agent_end` fired after planning completed, sending `/plan review` via `pi.sendUserMessage` re-entered the agent loop rather than running the command handler, endlessly re-triggering itself. In addition, plain `ExtensionContext` lacked `newSession` for fresh session implementation, and `PI_CODING_AGENT_DIR` needed tilde path expansion.

## Changes
- **plan-mode**: Invoke `showPlanReview(ctx, request)` directly upon plan completion in `agent_end`
- **plan-mode**: Clear `activePlanRequest` state immediately on review invocation and exit to prevent repeated triggers
- **plan-mode**: Retain command context from `/plan` invocation to support fresh session creation in `showPlanReview`
- **plan-mode**: Expand leading tildes in `getAgentDir` and plan file path resolvers
- **tests**: Add BDD scenario and regression test suite in `packages/plan-mode/tests/test_plan_mode.py`
- **changeset**: Add patch changeset for `@fradser/pi-plan-mode`

## Review cycle
2 comments triaged over 2 rounds, 2 adopted, 0 rejected; all CI green.
Full breakdown: https://github.com/FradSer/pi-packages/pull/31#issuecomment-5462134111

## Verification
- `python3 -m pytest packages/plan-mode/tests/ -q`: 23 passed
- `npx tsc --noEmit -p tsconfig.extensions.json`: 0 errors
- `pnpm --dir packages/plan-mode pack --dry-run`: verified

### CODE REVIEW

File: packages/plan-mode/src/index.ts [hunk 1]
```diff
@@ -518,14 +520,14 @@ export default function planMode(extensionApi: ExtensionAPI): void {
...
     planHandling = true;
     try {
       if (requiresWorkerResearch(planContent)) {
-        await runWorkerResearch(ctx, activePlanRequest, planContent);
+        await runWorkerResearch(ctx, request, planContent);
       } else {
-        // Send command message to trigger command handler with ExtensionCommandContext
-        // This allows newSession to be available for implement-fresh action
-        pi.sendUserMessage("/plan review");
+        await showPlanReview(ctx, request);
```
[the remark this answers is not in the document]
Comment (FradSer) [reply packages/plan-mode/src/index.ts:544]: Adopted

File: packages/plan-mode/src/config.ts [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
-import { getAgentDir } from "@earendil-works/pi-coding-agent";
 import { nonEmpty, parseModelRef } from "@fradser/pi-kit";
 
 export interface PlanModeConfig {
   provider?: string;
   model?: string;
 }
 
+function getAgentDir(): string {
+  return process.env.PI_CODING_AGENT_DIR ?? join(process.env.HOME ?? os.homedir(), ".pi", "agent");
```
[the remark this answers is not in the document]
Comment (FradSer) [reply packages/plan-mode/src/config.ts]: Adopted

**Comment by FradSer** [comment]
Summary of the review cycle on this PR.

## Changes made
- Prevent infinite `/plan review` loop by invoking `showPlanReview(ctx, request)` directly upon plan completion and clearing `activePlanRequest` immediately.
- Retain command context from `/plan` handler execution so subsequent agent-turn plan review calls can invoke `newSession` for fresh session implementation.
- Expand leading tilde (`~/`) paths in `getAgentDir` and plan file paths for consistent config/plan directory resolution.
- Added comprehensive BDD feature scenarios and regression test cases covering all edge cases.

## Review findings and resolution
- <Adopted> @code-terrier[bot] on packages/plan-mode/src/index.ts:530 — Auto-shown plan review cannot open a fresh session without command context → fixed in 8307a33.
- <Adopted> @code-terrier[bot] on packages/plan-mode/src/config.ts:12 — Reimplemented getAgentDir drops leading-tilde expansion of PI_CODING_AGENT_DIR → fixed in 8307a33.

All CI checks (Code Terrier Review, Devin Review) are green and no comments remain worth adopting. This PR is ready to merge.
